File: CS/cs/PersistentClasses/Order.cs
using DevExpress.Xpo;
using System;

namespace DXSample.PersistentClasses {
    public class Order : XPObject{
        public Order(Session session) : base(session) { }

        private string fOrderName;
        public string OrderName {
            get { return fOrderName; }
            set { SetPropertyValue<string>("OrderName", ref fOrderName, value); }
        }

        private DateTime fOrderDate;
        public DateTime OrderDate {
            get { return fOrderDate; }
            set { SetPropertyValue<DateTime>("OrderDate", ref fOrderDate, value); }
        }

        private Customer fCustomer;
        [Association("Customer-Orders")]
        public Customer Customer {
            get { return fCustomer; }
            set { SetPropertyValue<Customer>("Customer", ref fCustomer, value); }
        }
    }
}

File: VB/cs/PersistentClasses/Order.vb
Imports Microsoft.VisualBasic
Imports DevExpress.Xpo
Imports System

Namespace DXSample.PersistentClasses
	Public Class Order
		Inherits XPObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub

		Private fOrderName As String
		Public Property OrderName() As String
			Get
				Return fOrderName
			End Get
			Set(ByVal value As String)
				SetPropertyValue(Of String)("OrderName", fOrderName, value)
			End Set
		End Property

		Private fOrderDate As DateTime
		Public Property OrderDate() As DateTime
			Get
				Return fOrderDate
			End Get
			Set(ByVal value As DateTime)
				SetPropertyValue(Of DateTime)("OrderDate", fOrderDate, value)
			End Set
		End Property

		Private fCustomer As Customer
		<Association("Customer-Orders")> _
		Public Property Customer() As Customer
			Get
				Return fCustomer
			End Get
			Set(ByVal value As Customer)
				SetPropertyValue(Of Customer)("Customer", fCustomer, value)
			End Set
		End Property
	End Class
End Namespace